pkg/prometheus/common.go

```diff
@@ -594,3 +598,31 @@ func MakeProbes(
 
 	return startupProbe, readinessProbe, livenessProbe
 }
+
+// If a ServiceName is specified, check that the service exists in the namespace. If it does not exist, fail
+// the reconciliation.
+// Also, ensure that the Prometheus instance is selected by the service. If not, fail the reconciliation.
+func CheckCustomService(ctx context.Context, serviceName *string, prometheusNS string, prometheusName string, svcClient clientv1.ServiceInterface, selectorLabels map[string]string) error {
+	svc, err := svcClient.Get(ctx, *serviceName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("service %s/%s does not exist: %w", prometheusNS, *serviceName, err)
+		}
+
+		return err
+	}
+
+	// Get the user defined label selectors from the service.
+	svcSelector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: svc.Spec.Selector})
+	if err != nil {
+		return fmt.Errorf("failed to get label selector while synchronizing service: %w", err)
```

Comment:
IIUC this should never happen in real life.
```suggestion
		return fmt.Errorf("failed to convert service selector: %w", err)
```

Reply:
I have made the change, but could you elaborate a bit on that statement?

Reply:
I mean to say that the service shouldn't have an invalid label selector value (how would it be allowed by the API in the first place?). But it's better to handle the error in any case.